src/Domain/BlockLogicalCoordinates.cpp

```diff
@@ -133,6 +133,43 @@ block_logical_coordinates_single_point(
   return logical_point;
 }
 
+template <size_t Dim, typename Fr>
+BlockLogicalCoords<Dim> block_logical_coordinates_single_point(
+    const tnsr::I<double, Dim, Fr>& input_point, const Domain<Dim>& domain,
+    const double time, const domain::FunctionsOfTimeMap& functions_of_time,
+    const std::optional<gsl::not_null<std::vector<size_t>*>> block_order) {
+  // Check which block this point is in. Each point will be in one
+  // and only one block, unless it is on a shared boundary.  In that
+  // case, choose the first matching block (and this block will have
+  // the smallest block_id).
+  // In case a block_order is provided, it is no longer guaranteed that the
+  // block with the smallest block_id is chosen.
+  const size_t num_blocks = block_order.has_value()
+                                ? block_order.value()->size()
+                                : domain.blocks().size();
+  for (size_t i = 0; i < num_blocks; ++i) {
+    const size_t block_id =
+        block_order.has_value() ? (*block_order.value())[i] : i;
```

Comment:
`ASSERT(block_order.value()->size() <= domain.blocks().size())` to avoid a bad block order